spec.md

```diff
@@ -523,13 +518,295 @@ schema being used.
+and then the Platform is responsible to continue fetching any unknown
+`:component_path` URIs. Care ought to be taken by the platform to not fetch
+duplicate definitions.
+
+A Platform MAY ignore the `osb_v2:` scheme and just fetch `/v2/catalog/schemas`
+which will contain all sub-Schema documents. This reduces the complexity of the
+Platform but increases the potential payload size of the JSON Schemas.
+
+A Platform MUST fetch the root Schema document for any other `$ref` URIs found
+that do not match the `osb_v2:` scheme. If this happens the Platform has no use
+for the previously fetched sub-schemas and can just use the root schema. The
+Broker MUST provide the JSON Schema Object for the referenced URI in a subschema
+returned.
+
+Component Paths that use the scheme `osb_v2:` SHOULD be versioned, and the
+document behind the URI MUST NOT change after the catalog has been served to a
+Platform. 
+
+Platforms are able to leverage this JSON Schema by using or creating a library
+that supports pre-fetched JSON Schemas.
+
+Brokers MUST NOT allow duplicate JSON Pointers to be formed in any JSON Schema.
+
+A note about the scheme, at the time of this writing, there were no found client
+libraries that fully support the [JSON Schema](http://json-schema.org/) spec in
+the usage of URI for `$ref`. Only URL was supported in libraries. So as a
+compromise the `osb_v2:` scheme SHOULD be followed by three forward slashes
+(`///`). This indicates the authority is implied (authority is ''). 
+
+
+#### $ref Usage
+
+Brokers MAY leverage `$ref` and external JSON Schemas only if the JSON Schema
+references can be found within the [JSON Schemas](#json-schemas) API endpoint. $ref
+usage is assumed to be delegation (linking), not inclusion (replacement).
+Libraries that implement `$ref` content replacement result in adverse behaviors.
+
+For Brokers that support `$ref`:
+
+  * MUST respond with all JSON Schema documents for `/schemas` requests that are
+    referenced in the catalog.
+  * MAY provide `$ref`'s with the `osb_v2:` scheme, then
+    * MUST respond to `/schemas/:component_id` requests.
+
+### Request
+
+#### Route
+`GET /v2/catalog/schemas`
+
+#### Headers
+
+The following HTTP Headers are defined for this operation:
+
+| Header | Type | Description |
+| --- | --- | --- |
+| X-Broker-API-Version* | string | See [API Version Header](#api-version-header). |
+
+\* Headers with an asterisk are REQUIRED.
+
+#### cURL
+
+```
+$ curl http://username:password@service-broker-url/v2/catalog/schemas -H "X-Broker-API-Version: 2.13"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | MUST be returned upon successful processing of this request. The expected response body is below. |
+
+
+#### Body
+
+The root JSON Schema document is a complete collection of JSON Schemas
+leveraged by the Broker's catalog. The Platform will process this
+document before catalog `$ref` URIs map to definitions. The body MUST contain
```

Comment:
I don't know what: "The Platform will process this document before catalog `$ref` URIs map to definitions."
means. Can you elaborate?

Reply:
It means this only works with pre-fetched json schema support which is a more advanced Json Schema client feature.

Reply:
I realized that the platform does not have to process this. They could just pass the response along to the client UI and have that process it. So I reworked this to loosen it up.